### PR TITLE
feat(datahub-cli): datahub init & docs for it

### PR DESCRIPTION
**Pull request by gabe-lyons**
Add an easy way to configure datahub cli for remote instances.

```
(venv) ➜  datahub init
Configure which datahub instance to connect to
Enter your DataHub host [http://localhost:8080]: 
(Optional) Enter your DataHub access token []: 
Written to /Users/gabe/.datahubenv

(venv) ➜ datahub init
/Users/gabe/.datahubenv already exists. Overwrite? [y/N]: y
Configure which datahub instance to connect to
Enter your DataHub host [http://localhost:8080]: 
(Optional) Enter your DataHub access token []: 
Written to /Users/gabe/.datahubenv
```

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/src/datahub/cli/cli_utils.py [hunk 1]
```diff
@@ -30,6 +23,17 @@ class DatahubConfig(BaseModel):
     gms: GmsConfig
 
 
+def write_datahub_config(host: str, token: Optional[str]):
+    config = {
+        "gms": {
+            "server": host,
+            "token": token,
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/cli/cli_utils.py:30]: will this write "token": None if token is not provided?

File: docs/how/delete-metadata.md [hunk 2]
```diff
@@ -2,6 +2,18 @@
 
 There are a two ways to delete data from DataHub.
 
+
+## Configuring DataHub CLI
+
+The CLI will point to localhost DataHub by default. Running
+
+```aidl
```
Comment (kevinhu) [inline docs/how/delete-metadata.md:10]: Why is aidl used for syntax here?

Comment (gabe-lyons) [reply docs/how/delete-metadata.md:10]: Ah good question- intellij used this as the default and I didn't remove it

Comment (gabe-lyons) [reply docs/how/delete-metadata.md]: updated!

**Review by dj110919871 — APPROVED** [review]

File: metadata-ingestion/src/datahub/cli/cli_utils.py (hunk 1 above)
Comment (gabe-lyons) [reply metadata-ingestion/src/datahub/cli/cli_utils.py:30]: @shirshanka If token is None, then no Authorization header will be attached

File: docs/how/delete-metadata.md [hunk 3]
```diff
@@ -2,11 +2,23 @@
 
 There are a two ways to delete data from DataHub.
 
+
```
Comment (jjoyce0510) [inline docs/how/delete-metadata.md:5]: Would love a dedicated CLI doc, that this doc simply points to :)

File: metadata-ingestion/src/datahub/entrypoints.py [hunk 4]
```diff
@@ -58,6 +59,26 @@ def version() -> None:
...
 
+@datahub.command()
+def init() -> None:
+    """Configure which datahub instance to connect to"""
+    if os.path.isfile(DATAHUB_CONFIG_PATH):
+        click.confirm(f"{DATAHUB_CONFIG_PATH} already exists. Overwrite?", abort=True)
+
+    click.echo("Configure which datahub instance to connect to")
+    host = click.prompt(
+        "Enter your DataHub host", type=str, default="http://localhost:8080"
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/entrypoints.py:70]: Let's specify that this is GMS API server

**Review by jjoyce0510 — COMMENTED** [review]
Overall looks great!

File: metadata-ingestion/src/datahub/entrypoints.py (hunk 4 above)
Comment (gabe-lyons) [reply metadata-ingestion/src/datahub/entrypoints.py:70]: Some users might not know what GMS is, but I'll add a line in the docs to clarify this :)

File: docs/how/delete-metadata.md (hunk 3 above)
Comment (gabe-lyons) [reply docs/how/delete-metadata.md:5]: Yes!! I dont want to expand the scope of this PR, but completely agree.

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**